### DISCUSSION TITLE
OAuth 2.0 - Token expiration information

### DISCUSSION THREAD

**Discussion by stefanosalvucci** [fattureincloud/api#10] [Q&A] [answered]
Hello Fatture in Cloud team!
We're implementing V2 Oauth integration and I have a couple of questions:

- the access token will expire in 24h?
- when the refresh token expires?

Thank you

**Comment by valmoz** [comment, marked as answer]
Hi @stefanosalvucci  and welcome!
You can find the info you need here: https://developers.fattureincloud.it/docs/authentication/code-flow/vanilla-code#-token-types

Just to recap: the **access token** expires after 24 hours from its emission (the expires_in param is referred to this token), while the **refresh token** expires after 1 year since its last usage. So you can consider it perpetual if you don't make it expire.

Let us know if you have further questions.

Thanks
Mauro

**Reply by stefanosalvucci** [reply]
Thank you, I totally missed that page
